corehq/apps/linked_domain/dbaccessors.py

```diff
@@ -98,3 +98,19 @@ def _is_available_upstream_domain(candidate_name):
         return is_active_upstream_domain(candidate_name)
 
     return list({d.name for d in Domain.active_for_user(user) if _is_available_upstream_domain(d.name)})
+
+
+def get_domains_eligible_for_linked_apps(upstream_domain_name, user):
+    if domain_has_privilege(upstream_domain_name, RELEASE_MANAGEMENT):
+        upstream_membership = user.get_domain_membership(upstream_domain_name)
+        is_upstream_admin = upstream_membership.is_admin if upstream_membership else False
```

Comment:
Should this also check for superusers, who won't have a membership?

Reply:
Ah I didn't think about that. That does sound reasonable, and I should also make that change to the method that retrieves domains that are available to link, `get_available_domains_to_link`.